Add support for BINDNOW_FLAGS to build kgrantpty and kpac_dhcp_helper with
non-lazy bindings, to improve security and fix installation on
FEATURES=stricter.
See bug #113937.

Requires the ebuild to export BINDNOW_FLAGS="$(bindnow-flags)".

Index: kdelibs-3.5.0/kdecore/Makefile.am
===================================================================
--- kdelibs-3.5.0.orig/kdecore/Makefile.am
+++ kdelibs-3.5.0/kdecore/Makefile.am
@@ -169,7 +169,7 @@ kde_config_LDFLAGS = $(KDE_RPATH) $(KDE_
 
 kgrantpty_SOURCES = kgrantpty.c
 kgrantpty_CFLAGS= $(KDE_USE_FPIE)
-kgrantpty_LDFLAGS = $(KDE_USE_PIE) $(KDE_RPATH) $(all_libraries)
+kgrantpty_LDFLAGS = $(KDE_USE_PIE) $(KDE_RPATH) $(all_libraries) $(BINDNOW_FLAGS)
 kgrantpty_LDADD = ./libkdefakes.la
 
 # kgrantpty needs to be installed setuid root
Index: kdelibs-3.5.0/kio/misc/kpac/Makefile.am
===================================================================
--- kdelibs-3.5.0.orig/kio/misc/kpac/Makefile.am
+++ kdelibs-3.5.0/kio/misc/kpac/Makefile.am
@@ -14,7 +14,7 @@ kded_proxyscout_la_LIBADD = $(LIB_KIO) $
 
 kpac_dhcp_helper_SOURCES = kpac_dhcp_helper.c
 kpac_dhcp_helper_CFLAGS = $(KDE_USE_FPIE)
-kpac_dhcp_helper_LDFLAGS = $(KDE_USE_PIE)
+kpac_dhcp_helper_LDFLAGS = $(KDE_USE_PIE) $(BINDNOW_FLAGS)
 kpac_dhcp_helper_LDADD = $(LIBSOCKET)
 
 noinst_HEADERS = proxyscout.h script.h downloader.h discovery.h
